adagrams/game.py

```diff
@@ -1,11 +1,80 @@
+import random
+
 def draw_letters():
-    pass
+    
+    letters   = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 
+    'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2, 
+    'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1, 
+    'Y': 2, 'Z': 1 }
+
+    pieces = []
+    # number of pieces in letter pool
+    count = 98
+
+    for i in range(10):
+        rand_piece = random.randrange(1, count+1)
+        key = weighted(letters, rand_piece)
+        pieces.append(key)
+        
+        letters[key] -=1
+        if letters[key] == 0:
+            letters.pop(key)
+        count -= 1
+    return pieces
 
+def weighted(dict,random_piece):
+    for char, weight in dict.items():
+            random_piece -= weight
+            if random_piece <= 0:
+                return char
+            
 def uses_available_letters(word, letter_bank):
-    pass
+    # check if they can use the word they inputed
+    new_letter_bank = []
+    for l in letter_bank:
+        new_letter_bank.append(l.upper())
```

Comment:
Nice job for considering if the `letter_bank` is not capitalized, though you don't necessarily need to consider this since `letter_bank` is the return value of `draw_letters`. With how you implemented `draw_letters` it will never return any lowercase values.